core/src/test/java/cucumber/runtime/RuntimeGlueTest.java

```diff
@@ -60,4 +78,63 @@ public void removes_glue_that_is_scenario_scoped() {
         assertEquals(0, glue.beforeHooks.size());
         assertEquals(0, glue.afterHooks.size());
     }
+
+    @Test
+    public void step_definition_match() {
```

Comment:
This should be a name that describes what the test is testing. This may require splitting up this test case into separate smaller test cases.